lib/rex/parser/fs/ntfs.rb

```diff
@@ -0,0 +1,257 @@
+          return parse_index_list(index_record, attributes[INDEX_ALLOCATION_ID])
+        else
+          return parse_index_list(index_record, "")
+        end
+      end
+
+      def cluster_from_attribute_non_resident(attribute, cluster_num = 0, size_max = ((2**31) - 1))
+        lowvcn = attribute[16, 8].unpack("Q<")[0]
+        highvcn = attribute[24, 8].unpack("Q<")[0]
+        offset = attribute[32, 2].unpack("v")[0]
+        real_size = attribute[48, 8].unpack("Q<")[0]
+        attribut = ""
+        run_list_num = lowvcn
+        old_offset = 0
+
+        while run_list_num <= highvcn
+          first_runlist_byte = attribute[offset].ord
+          run_offset_size = first_runlist_byte >> 4
+          run_length_size = first_runlist_byte & 15
+          run_length = attribute[offset + 1, run_length_size]
+          run_length += "\x00" * (8 - run_length_size)
+          run_length = run_length.unpack("Q<")[0]
+
+          offset_run_offset = offset + 1 + run_length_size
+          run_offset = attribute[offset_run_offset, run_offset_size]
+          if run_offset[-1].ord & 128 == 128
+            run_offset += "\xFF" * (8 - run_offset_size)
+          else
+            run_offset += "\x00" * (8 - run_offset_size)
+          end
+          run_offset = run_offset.unpack("q<")[0]
+
+          size_wanted = [run_length * @bytes_per_cluster, size_max - attribut.length].min
+
+          if cluster_num + (size_wanted / @bytes_per_cluster) >= run_list_num && (cluster_num < run_length + run_list_num)
+            run_list_offset_in_cluster = run_offset + old_offset + [cluster_num - run_list_num, 0].max
+            run_list_offset = (run_list_offset_in_cluster) * @bytes_per_cluster
+            run_list_offset = run_list_offset.to_i
+            @file_handler.seek(run_list_offset)
+
+            data = ""
+            while data.length < size_wanted
+              data << @file_handler.read(size_wanted - data.length)
+            end
+            attribut += data
+          end
+          offset += run_offset_size + run_length_size + 1
+          run_list_num += run_length
+          old_offset = run_offset
+        end
+        attribut = attribut[0, real_size]
+        attribut
+      end
+
+      #
+      # return the attribute list from the MFT record
+      # deal with resident and non resident attributes (but not $DATA due to perforemence issue)
+      #
+      def mft_record_attribute(mft_record)
+        attribute_list_offset = mft_record[20, 2].unpack("C")[0]
+        curs = attribute_list_offset
+        attribute_identifier = mft_record[curs, 4].unpack("V")[0]
+        res = {}
+        while attribute_identifier != 0xFFFFFFFF
+          # attribute_size=mft_record[curs + 4, 4].unpack("V")[0]
+          # should be on 4 bytes but doesnt work
+          attribute_size = mft_record[curs + 4, 2].unpack("v")[0]
+          #print_debug("attribute_size: #{attribute_size}, attribute_identifier: #{attribute_identifier}")
```

Comment:
Don't leave commented-out code in the source. If it's something that might be useful while debugging, try vprint_debug, which will only output when verbose.